tests/test_preprints.py

```diff
@@ -2584,8 +2584,8 @@ def test_new_version_pre_mod_submit_and_reject(self, preprint_pre_mod, creator,
         assert new_version.machine_state == ReviewStates.REJECTED.value
         assert new_version.versioned_guids.first().is_rejected is True
         guid_obj = new_version.get_guid()
-        assert guid_obj.object_id == preprint_pre_mod.pk
-        assert guid_obj.referent == preprint_pre_mod
+        assert guid_obj.object_id == new_version.pk
+        assert guid_obj.referent == new_version
```

Comment:
This is another case that I forgot to ask during my first review. And this is probably a product question too.

* Can you verify that rejected  preprints will keep the version number? (Rejection only happens for pre-mod).
* Does product really want the behavior that non-version guid leads to rejected preprints? Or should we change the version to point back to an earlier "latest version" upon rejection?